### ISSUE TITLE
umount force considered harmfull

### ISSUE THREAD

**Issue by hekmon** [hekmon/rclonemount#1]
Following https://github.com/rclone/rclone/issues/4595#issuecomment-918189138

> `MNT_FORCE` is actually dangerous:
> 
> ```
> MNT_FORCE (since Linux 2.1.116)
>               Ask the filesystem to abort pending requests before
>               attempting the unmount.  This may allow the unmount to
>               complete without waiting for an inaccessible server, but
>               could cause data loss.  If, after aborting requests, some
>               processes still have active references to the filesystem,
>               the unmount will still fail.
> ```
> 
> While `MNT_DETACH`:
> 
> ```
> Perform a lazy unmount: make the mount unavailable for new
>               accesses, immediately disconnect the filesystem and all
>               filesystems mounted below it from each other and from the
>               mount table, and actually perform the unmount when the
>               mount ceases to be busy.
> ```
> 
> So basically it detaches the file system immediately for new accesses and as soon as rclone process dies, hence any possible open descriptors are released, unmount completes cleanly. Have never seen it fail in any of my use cases (eg gocryptfs encrypted home directories etc.) and is safer for data.

**Comment by hekmon** [comment]
force is always risky, but here it is not used to perform the actual/regular umount which is done by rclone during stopping after sig int in Exec=. This umount is only here to be absolutly sure the mount point will be reusable in case of a failure during the rclone stop (mostly hang for the system while rclone exits).

From the umount man page:
```
       -l, --lazy
              Lazy unmount.  Detach the filesystem from the file hierarchy now, and clean up all references to this filesystem as soon as it is not busy anymore.

              A system reboot would be expected in near future if you're going to use this option for network filesystem or local filesystem with submounts.  The recommended use-case for umount -l is to prevent hangs on shutdown due to an unreachable network share where a normal umount will hang  due  to  a
              downed server or a network partition. Remounts of the share will not be possible.
```

To my understanding this won't do for this specific usecase I want to protect from as even if the lazy mode launch an async umount, for a use case of unit restart with a umount hang, the mount point will not be available for the rclone restart just after during the service restart.

TL;DR: this umount ExecStartPost= is not the actual umount (rclone stop is) it is here only to be 100% sure the mount point will be reusable in case of service restart is the umount is actually stuck for whatever reason (and again this is extremly rare).

**Comment by Nodens-** [comment]
That paragraph regarding system reboot is poorly phrased as it pertains to specific conditions with sub-mounts with active processes and zombie processes.
In this particular use case, the only situation where it won't unmount is if rclone has bugged out so hard that it ends up a zombie process, hence unkillable until reboot, with open file descriptors. But in that case `umount -f` will simply fail to umount due to active references, so there's nothing that can be done for this specific scenario (although in this specific scenario `-zu` will remove the reboot/shutdown delay, with unavailable mount point until reboot of course, while `-f` will not, it will just fail and timeout normally). For all other cases the mount point will be available when the process restarts.

**Comment by Nodens-** [comment]
For simplicity's sake the mount point is not available if there are file descriptors/locks kernel side that are not released as the unmount process never actually completes, you are left with the mount point just detached from the filesystem but still in use by the kernel. But in this case force fails as well with the only difference being having blocking/timeout behavior. That's the main difference here. For other cases, force can cause data loss while lazy will eventually unmount properly but in my opinion it should not really be used for mount points that have nesting or multiple processes working on them that are possibly not terminated during the unmount process.
